src/com/owncloud/android/operations/UploadFileOperation.java

```diff
@@ -372,6 +372,9 @@ protected RemoteOperationResult run(OwnCloudClient client) {
             if (temporalFile != null && !originalFile.equals(temporalFile)) {
                 temporalFile.delete();
             }
+            if (result == null){
+                return new RemoteOperationResult(false, 404, null);
```

Comment:
The only way I can see that something could lead to ```result == null``` is that a ```java.lang.Error```occurred... what is pretty bad itself